src/apps/reservation-list/modal/reservation-details/physical-list-details.tsx

```diff
@@ -94,16 +94,18 @@ const PhysicalListDetails: FC<PhysicalListDetailsProps & MaterialProps> = ({
       selectedExpiryDate = getFutureDateString(selectedInterest);
     }
 
+    const reservationsChanges = reservationIds.map((reservationId) => {
+      return {
+        expiryDate: selectedExpiryDate,
+        pickupBranch: selectedBranch,
+        reservationId
+      };
+    });
+
     mutate(
       {
         data: {
-          reservations: [
-            {
-              expiryDate: selectedExpiryDate,
-              pickupBranch: selectedBranch,
-              reservationId
-            }
-          ]
+          reservations: reservationsChanges
```

Comment:
This is the second key change. Instead of applying changes to a single (physical) reservation id we now create an array of multiple reservations to update if this is a parallel reservation.